pkg/commands/products/root.go

```diff
@@ -0,0 +1,189 @@
+package products
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/fastly/go-fastly/v8/fastly"
+
+	"github.com/fastly/cli/pkg/cmd"
+	fsterr "github.com/fastly/cli/pkg/errors"
+	"github.com/fastly/cli/pkg/global"
+	"github.com/fastly/cli/pkg/manifest"
+	"github.com/fastly/cli/pkg/text"
+)
+
+// RootCommand is the parent command for all subcommands in this package.
+// It should be installed under the primary root command.
+type RootCommand struct {
+	cmd.Base
+	cmd.JSONOutput
+
+	disableProduct string
+	enableProduct  string
+	manifest       manifest.Data
+	serviceName    cmd.OptionalServiceNameID
+}
+
+// ProductEnablementOptions is a list of products that can be enabled/disabled.
+var ProductEnablementOptions = []string{
+	"brotli_compression",
+	"domain_inspector",
+	"fanout",
+	"image_optimizer",
+	"origin_inspector",
+	"websockets",
+}
+
+// NewRootCommand returns a new command registered in the parent.
+func NewRootCommand(parent cmd.Registerer, g *global.Data, m manifest.Data) *RootCommand {
+	var c RootCommand
+	c.Globals = g
+	c.manifest = m
+	c.CmdClause = parent.Command("products", "Enable, disable, and check the enablement status of products on your services")
+
+	// Optional.
+	c.CmdClause.Flag("disable", "Disable product").HintOptions(ProductEnablementOptions...).EnumVar(&c.disableProduct, ProductEnablementOptions...)
+	c.CmdClause.Flag("enable", "Enable product").HintOptions(ProductEnablementOptions...).EnumVar(&c.enableProduct, ProductEnablementOptions...)
+	c.RegisterFlagBool(c.JSONFlag()) // --json
+	c.RegisterFlag(cmd.StringFlagOpts{
+		Name:        cmd.FlagServiceIDName,
+		Description: cmd.FlagServiceIDDesc,
+		Dst:         &c.manifest.Flag.ServiceID,
+		Short:       's',
+	})
+	c.RegisterFlag(cmd.StringFlagOpts{
+		Action:      c.serviceName.Set,
+		Name:        cmd.FlagServiceName,
+		Description: cmd.FlagServiceDesc,
+		Dst:         &c.serviceName.Value,
+	})
+	return &c
+}
+
+// Exec implements the command interface.
+func (c *RootCommand) Exec(_ io.Reader, out io.Writer) error {
+	ac := c.Globals.APIClient
+
+	if c.enableProduct != "" && c.disableProduct != "" {
+		return fsterr.ErrInvalidProductEnablementFlagCombo
+	}
+
+	if c.Globals.Verbose() && c.JSONOutput.Enabled {
+		return fsterr.ErrInvalidVerboseJSONCombo
+	}
+
+	serviceID, _, _, err := cmd.ServiceID(c.serviceName, c.manifest, c.Globals.APIClient, c.Globals.ErrLog)
+	if err != nil {
+		return fmt.Errorf("failed to identify Service ID: %w", err)
+	}
+
+	if c.enableProduct != "" {
+		if _, err := ac.EnableProduct(&fastly.ProductEnablementInput{
+			ProductID: identifyProduct(c.enableProduct),
```

Comment:
Wondering if we should check for `fastly.ProductUndefined` before making the API call?

Reply:
We already use the `EnumVar` so we know we're going to be getting a value that corresponds to the switch statement. I guess checking for `ProductUndefined` would only be useful if we're concerned the switch statement had a typo (which is completely possible, so I'll look at adding a check 👍🏻 )

Reply:
Got it -- my concern was avoiding unnecessary API calls. But if that's handled within the client, then it's not a concern.

Reply:
No worries. I've added it because the concern actually comes from us translating the enum flag's value (e.g. "brotli_compression") into an actual Go iota/enum, and at that point we could well have a typo hardcoded in the code which no one notices until runtime 😬 so I think you're right and it's best to throw that check in just to be on the safe side.